src/server/routes/auth.ts

```diff
@@ -1,13 +1,46 @@
 import { Router } from 'express';
 import * as passport from 'passport';
 import { Environment } from '../config/Environment';
+import * as bodyParser from 'body-parser';
+
+const postParser = bodyParser.urlencoded({
+    extended: false,
+});
 
 const router: Router = Router();
 
 const failureRedirect = '/';
 const successRedirect = '/';
 
+router.get('/user_data', (req, res) => {
+    if (req.user === undefined) {
+        res.json({});
+    } else {
+        res.json({
+            userid: req.user.userid,
+            email: req.user.email,
+            name: req.user.name,
+            userclass: req.user.userclass,
+        });
+    }
+});
+
+router.get('/user_data/class', (req, res) => {
+    if (req.user === undefined) {
+        res.json({
+            a: 4,
+        });
+    } else {
+        res.json({
+            a: req.user.userclass,
+        });
+    }
+});
 
+router.get('/logout', (req, res) => {
+    req.logout();
+    res.redirect('/login');
```

Comment:
isn't this going to be called with fetch(), like every other part of the API? redirect doesn't make sense then